### ISSUE TITLE
copy directory and rename

### ISSUE THREAD

**Issue by HongxuChen** [webdesus/fs_extra#5]
I hope fs_extra's copy for directory can simulate `cp` command. e.g., `cp d1 d2`, if `d2` does not exist it copy the whole directory `d1` to `d2` (in this ase `d1` and `d2` in same folder with same substructure); if `d2` exists, copy `d1` into `d2` (as what has been implemented). The default behavior can be changed by modifying `CopyOptions`.

**Comment by webdesus** [comment]
Sorry I don't understand your question. You really can use `CopyOptions` for realise anything behaviours.

**Comment by HongxuChen** [comment]
@webdesus I mean

(1) When the specified directory (`d2`) doesn't exists:

``` tree .
.
└── d1
   ├── f1
   └── f2
```

And after `cp -r d1 d2`, it will mirror `d1` at the same root directory, with the target directory's name changed to `d2`.

```
.
├── d1
│  ├── f1
│  └── f2
└── d2
   ├── f1
   └── f2
```

(2) But if `d2` exists,

```
.
├── d1
│  ├── f1
│  └── f2
└── d2
```

`cp -r d1 d2` defaults to copy the whole directory `d1` inside `d2`,

```
.
├── d1
│  ├── f1
│  └── f2
└── d2
   └── d1
      ├── f1
      └── f2
```

**Comment by webdesus** [comment]
Nice feauter. I think will be better add new attribute in `CopyOptions` by name `place_inside`. Unfortunately I am now busy and will be able to realize this opportunity ~2-3 weeks. If you wanna help I will be glad accept yours PR. 😊

**Comment by HongxuChen** [comment]
@webdesus I'll do if I have time (but sadly I'm also occupied with other affairs these days).

**Comment by webdesus** [comment]
Enhancement will be access in 1.1.0 and hight version. Use copy_inside varible of CopyOptions object. Update will be access on this week.

**Comment by webdesus** [comment]
Published new version `fs_extra` **1.1.0**. For this functionality use `copy_inside` option of `CopyOptions`.

Sorry for so long release, i was hard busy.

**Comment by HongxuChen** [comment]
Thanks, really appreciate it!
